doc/monitoring_containers/monitoring_containers.md

```diff
@@ -0,0 +1,278 @@
+
+```bash
+check program container_memory_lldp with path "/usr/bin/memory_checker lldp 104857600"
+    if status != 0 then alert
+```
+
+### 2.2.3 Auto-restart Docker Container
+The design principle behind this auto-restart feature is docker containers can be automatically shut down and
+restarted if one of critical processes running in the container exits unexpectedly. Restarting
+the entire container ensures that configuration is reloaded and all processes in the container
+get restarted, thus increasing the likelihood of entering a healthy state.
+
+Currently SONiC used supervisord system tool to manage the processes in each
+docker container. Actually auto-restarting docker container is based on the process 
+monitoring/notification framework. Specifically
+if the state of process changes for example from running to exited,
+an event notification `PROCESS_STATE_EXITED` will be emitted by supervisord.
+This event will be received by event listener. The event listener determines if the process is
+critical process and whether it exited unexpectedly. If both of
+these conditions are true, the event listener will kill the supervisord process. Since supervisord
+runs as PID 1 inside the containers, when supervisord exits, the container will stop. When the
+container stops, the systemd service which manages the container will also stop, but it is
+configured to automatically restart the service, thus it will restart the container.
+
+We also introduced a configuration option which can enable or disable this auto-restart feature
+dynamically according to the requirement of users. In detail, we created a table 
+named `CONTAINER_FEATURE` in Config_DB and this table includes the status of
+auto-restart feature for each docker container. Users can easily use CLI to
+check and configure the corresponding docker container status.
+
+
+#### 2.2.3.1 CLI (and usage example)
+The CLI tool will provide the following functionality:
+1. Show current status of auto-restart feature for docker containers.
+2. Configure the status of a specific docker container.
+
+##### 2.2.3.1.1 Show the Status of Auto-restart
+```
+admin@sonic:~$ show container feature autorestart
+Container Name         Status 
+--------------------  --------
+database               disabled
+lldp                   disabled
+radv                   disabled
+pmon                   disabled
+sflow                  enabled
+snmp                   enabled
+telemetry              enabled
+bgp                    disabled
+dhcp_relay             disabled
+rest-api               enabled
+teamd                  disabled
+syncd                  enabled
+swss                   disabled
+```
+
+##### 2.2.3.1.2 Configure the Status of Auto-restart
+```
+admin@sonic:~$ sudo config container feature autorestart database enabled
+```
+
+
+##### 2.2.3.1.3 CONTAINER_FEATURE Table
+Example:
+```
+{
+    "CONTAINER_FEATURE": {
+        "database": {
+            "auto_restart": "enabled",
```

Comment:
You should also add the proposed "high_mem_alert", "high_cpu_alert" and "high_disk_alert" features here. Also, add a section to describe our proposal for configuring these thresholds.